Civi/Api4/Utils/CoreUtil.php

```diff
@@ -73,4 +73,34 @@ public static function getApiNameFromTableName($tableName) {
     return $entityName;
   }
 
+  /**
+   * Given an expression like "activity_date_time IS previous.month", rewrite to
+   * an equivalent expression "activity_date_time BETWEEN {start-of-previous-month} AND {end-of-previous-month}.
+   *
+   * @param string $fieldName
+   *   Ex: 'activity_date_time'
+   * @param string $operator
+   *   Ex: 'IS' or 'IS NOT'
+   * @param string $criteria
+   *   Ex: ['IS' => 'previous.month']
+   * @return array|NULL
+   *   Array(string $newOperator, mixed $newCriteria).
+   *   Ex: ['BETWEEN' => ['2020-05-01', '2020-06-01']]
+   */
+  public static function rewriteIsCriteria($fieldName, $operator, $criteria) {
+    $affirm = ($operator == 'IS');
+    if ($criteria === 'null' || $criteria === 'NULL') {
+      return $affirm ? ['IS NULL' => ''] : ['IS NOT NULL' => ''];
+    }
+
+    $relDateFilters = \CRM_Core_OptionGroup::values('relative_date_filters');
+    if (isset($relDateFilters[$criteria])) {
+      list ($dateFrom, $dateTo) = \CRM_Utils_Date::getFromTo($criteria, NULL, NULL);
+      $newOp = $affirm ? 'BETWEEN' : 'NOT BETWEEN';
+      return [$newOp => [\CRM_Utils_Date::mysqlToIso($dateFrom), \CRM_Utils_Date::mysqlToIso($dateTo)]];
+    }
+
+    return NULL;
```

Comment:
Instead of returning NULL wouldn't we default to replacing IS with = and IS NOT with !=?

Reply:
In keeping with the comment above... the idea in the current PR is to avoid ambiguity by strictly spliting `=` (match by value) and `IS` (match by logical symbol).  Returning `NULL` here means that it will throw an exception if the logical-symbol is unrecognized.

If `IS` were to fallback to `=`, then `IS` would have ambiguity on values like `'this.year'`.